pkg/api/validation/schema.go

```diff
@@ -297,6 +297,10 @@ func (s *SwaggerSchema) isGenericArray(p swagger.ModelProperty) bool {
 var versionRegexp = regexp.MustCompile(`^v.+\..*`)
 
 func (s *SwaggerSchema) validateField(value interface{}, fieldName, fieldType string, fieldDetails *swagger.ModelProperty) []error {
+	allErrs := []error{}
+	if reflect.TypeOf(value) == nil {
+		return append(allErrs, fmt.Errorf("unexpected nil value for field %v", fieldName))
```

Comment:
Hi @nikhiljindal , thanks for the review.

So I tried returning a success (empty error list) then ran a hacked version of "blog-logging". The user gets a rather poor error:

```
./kubectl.sh create -f ../examples/blog-logging/counter-pod.yaml 
unable to decode "../examples/blog-logging/counter-pod.yaml": [pos 94]: json: expect char '"' but got char 'n'
```
Instead of the following:
```
./kubectl.sh create -f ../examples/blog-logging/counter-pod.yaml 
error validating "../examples/blog-logging/counter-pod.yaml": error validating data: unexpected nil value for field spec.containers[0].args[0]; if you choose to ignore these errors, turn validation off with --validate=false
```